src/main/java/com/connexta/poller/service/Poller.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class Poller {
+
+  private final StatusService statusService;
+  private final BlockingQueue<URI> transferQueue;
+  private ExecutorService executor;
+  private AtomicBoolean isStarted;
+
+  public Poller(StatusService statusService, BlockingQueue<URI> transferQueue) {
+    this.statusService = statusService;
+    this.transferQueue = transferQueue;
+    isStarted = new AtomicBoolean(false);
+  }
+
+  public void start() {
+
+    // Prevent starting more than once
+    if (isStarted.compareAndSet(false, true)) {
+
+      // The stop() method sets this var to null to allow restarts
+      if (executor == null) {
+        // Nothing magical about the number of threads. We can tweak it after we see how it is
+        // working.
+        executor = Executors.newFixedThreadPool(32);
+      }
+
+      // Start consuming work
+      executor.submit(this::beginWorking);
+    }
+  }
+
+  private void beginWorking() {
+    URI statusUri = null;
+    while (!executor.isTerminated()) {
+      try {
+        statusUri = transferQueue.take();
+      } catch (InterruptedException e) {
+
+        isStarted.set(false);
+        // Reset interrupt
+        Thread.currentThread().isInterrupted();
```

Comment:
I think it is right. It looks weird, but isInterrupted() resets the the interrupt flag on the thread. I don't think I need to re-throw interrupt or interrupt the current thread. Exiting the beginWorking() method should end the thread because it is the bottom of the stack.

Reply:
The javadoc says
> The interrupted status of the thread is unaffected by this method.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Thread.html#isInterrupted()

https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java says
```java
} catch (InterruptedException e) {
    Thread.currentThread().interrupt();
```

If we don't want to re-interrupt the thread, then I think `Thread.currentThread().isInterrupted();` is meaningless.

Reply:
Updated